tracer/test/Datadog.Trace.Tests/Logging/DatadogLoggingFactoryTests.cs

```diff
@@ -0,0 +1,139 @@
+// </copyright>
+
+using System;
+using System.IO;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Logging;
+
+public class DatadogLoggingFactoryTests
+{
+    public class FileLoggingConfiguration
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("C:/")]
+        [InlineData("/var/root")]
+        public void UsesLogDirectoryWhenItExists(string obsoleteLogDirectory)
+        {
+            // will always exist
+            var logDirectory = Environment.CurrentDirectory;
+
+            var source = new NameValueConfigurationSource(
+                new()
+                {
+                    { ConfigurationKeys.LogDirectory, logDirectory },
+#pragma warning disable CS0618
+                    { ConfigurationKeys.ProfilerLogPath, obsoleteLogDirectory },
+#pragma warning restore CS0618
+                });
+
+            var config = DatadogLoggingFactory.GetConfiguration(source);
+            config.File.HasValue.Should().BeTrue();
+            config.File?.LogDirectory.Should().Be(logDirectory);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void UsesObsoleteLogDirectoryWhenAvailable(string logDirectory)
+        {
+            // will always exist
+            var obsoleteLogDirectory = Environment.CurrentDirectory;
+            var obsoleteLogPath = $"{obsoleteLogDirectory}{Path.DirectorySeparatorChar}{Guid.NewGuid()}.log";
+
+            var source = new NameValueConfigurationSource(
+                new()
+                {
+                    { ConfigurationKeys.LogDirectory, logDirectory },
+#pragma warning disable CS0618
+                    { ConfigurationKeys.ProfilerLogPath, obsoleteLogPath },
+#pragma warning restore CS0618
+                });
+
+            var config = DatadogLoggingFactory.GetConfiguration(source);
+            config.File.HasValue.Should().BeTrue();
+            config.File?.LogDirectory.Should().Be(obsoleteLogDirectory);
+        }
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData("", null)]
+        [InlineData(null, "")]
+        [InlineData("", "")]
+        public void UsesEnvironmentFallBackWhenBothNull(string logDirectory, string obsoleteLogDirectory)
+        {
+            var source = new NameValueConfigurationSource(
+                new()
+                {
+                    { ConfigurationKeys.LogDirectory, logDirectory },
+#pragma warning disable CS0618
+                    { ConfigurationKeys.ProfilerLogPath, obsoleteLogDirectory },
+#pragma warning restore CS0618
+                });
+
+            var config = DatadogLoggingFactory.GetConfiguration(source);
+            config.File.HasValue.Should().BeTrue();
+            config.File?.LogDirectory.Should().NotBeNullOrWhiteSpace();
```

Comment:
What is this falling back on? Is this repeatable in both CI and dev machines?

Reply:
This is falling back on our "default" folder, based on the environment (i.e. Windows/Linux). It may be different between machines, but will always have a value, even if it's the temp directory.

"Environment" here isn't the best choice of phrase; it's nothing to do with falling back on environment variables 😄